src/system_utils.jl

```diff
@@ -21,3 +93,99 @@ function (sm::SystemMap{DT})(u0, p) where {DT}
                       p = convert(typeof(sm.prob.p), p))
     solve(prob, sm.args...; sm.kwargs...)
 end
+
+function ExpectationProblem(sm::SystemMap, g, h, d; nout = 1)
+    ExpectationProblem(sm, g, h, d,
+                       ArrayPartition(deepcopy(sm.prob.u0), deepcopy(sm.prob.p)),
+                       nout)
+end
+
+"""
+```julia
+ProcessNoiseSystemMap(prob, args...; kwargs...)
+```
+
+Representation of a system solution map for a given `prob::DEProblem`. `args` and `kwargs`
+are forwarded to the equation solver.
+"""
+struct ProcessNoiseSystemMap{DT <: DiffEqBase.DEProblem, A, K} <: AbstractSystemMap
+    prob::DT
+    n::Int
```

Comment:
Is the only difference between the `SystemMap` and `ProcessNoiseSystemMap` the `n` parameter? Would it make any sense to just add `n` to `SystemMap`, dispatch on `SystemMap{DT}`, and not use `n` for non-`SDEProblem`?

Essentially, can we eliminate the need for multiple `XXXSystemMap` types.